sdk/tests/client/input_selection/foundry_outputs.rs

```diff
@@ -688,73 +711,54 @@ fn simple_foundry_transition_basic_not_needed_with_remainder() {
-    ]);
+    let mut inputs = build_inputs(vec![Foundry(
+        1_000_000,
+        alias_id_1,
+        1,
+        SimpleTokenScheme::new(U256::from(100), U256::from(0), U256::from(200)).unwrap(),
+        Some(vec![(&token_id.to_string(), 100)]),
+    )]);
+    let alias_output = AliasOutputBuilder::new_with_amount(2_000_000, alias_id_1)
+        .add_unlock_condition(StateControllerAddressUnlockCondition::new(
+            Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
+        ))
+        .add_unlock_condition(GovernorAddressUnlockCondition::new(
+            Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
+        ))
+        .with_foundry_counter(1)
+        .with_state_index(1)
+        .finish_output(protocol_parameters.token_supply())
+        .unwrap();
+    inputs.push(InputSigningData {
+        output: alias_output,
+        output_metadata: OutputMetadata::new(rand_block_id(), rand_output_id(), false, None, None, None, 0, 0, 0),
+        chain: None,
+    });
+
     let outputs = build_outputs(vec![Foundry(
         1_000_000,
         alias_id_1,
-        0,
+        1,
         SimpleTokenScheme::new(U256::from(120), U256::from(0), U256::from(200)).unwrap(),
         Some(vec![(&token_id.to_string(), 110)]),
     )]);
 
     let selected = InputSelection::new(
         inputs.clone(),
-        outputs.clone(),
+        outputs,
         addresses(vec![BECH32_ADDRESS_ED25519_0]),
         protocol_parameters,
     )
     .burn(Burn::new().add_native_token(token_id, 10))
-    .select()
-    .unwrap();
+    .select();
 
-    assert!(unsorted_eq(&selected.inputs, &inputs));
-    assert_eq!(selected.outputs.len(), 2);
-    assert!(selected.outputs.contains(&outputs[0]));
-    selected.outputs.iter().for_each(|output| {
-        if !outputs.contains(output) {
-            if output.is_alias() {
-                assert_eq!(output.amount(), 2_000_000);
-                assert_eq!(output.as_alias().native_tokens().len(), 0);
-                assert_eq!(output.as_alias().state_index(), 1);
-                assert_eq!(*output.as_alias().alias_id(), alias_id_1);
-                assert_eq!(output.as_alias().unlock_conditions().len(), 2);
-                assert_eq!(output.as_alias().features().len(), 0);
-                assert_eq!(output.as_alias().immutable_features().len(), 0);
-                assert_eq!(
-                    *output.as_alias().state_controller_address(),
-                    Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap()
-                );
-                assert_eq!(
-                    *output.as_alias().governor_address(),
-                    Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap()
-                );
-            } else {
-                panic!("unexpected output type")
-            }
-        }
-    });
+    assert!(matches!(
+        selected,
+        Err(Error::UnfulfillableRequirement(Requirement::Foundry(id))) if id == foundry_id
+    ));
```

Comment:
Is it not possible to mint and burn at the same time?

Reply:
Not with the same token id `The presence and validation of the foundry outputs of the native tokens determines whether the tokens are burned (removed from the ledger) or melted in the foundry.` https://github.com/iotaledger/tips/blob/main/tips/TIP-0018/tip-0018.md#additional-semantic-transaction-validation-rules